CONFIGURACIÓN DE LIBRERÍAS Y PARÁMETROS

In [77]:
# ===================================================================
# PASO 1: CONFIGURACIÓN E INSTALACIÓN DE LIBRERÍAS
# ===================================================================

import pandas as pd
import requests
import sqlite3
pd.set_option('display.max_columns', None) ## muestra todos los datos
pd.set_option('display.max_rows', None)
print("✅ Librerías importadas.")

✅ Librerías importadas.


EXTRACCIÓN DE LOS DATOS

In [2]:
# ===================================================================
# PASO 2: EXTRACCIÓN (EXTRACT) DE LOS DATOS
# ===================================================================

# La plataforma datos.gov.co usa la API de Socrata. Podemos usarla para
# descargar los datos directamente, lo que es más eficiente que bajar un CSV.
# Aumentamos el límite para traer más filas (ajusta si es necesario).
api_url = "https://www.datos.gov.co/resource/nudc-7mev.json?$limit=50000"

print(f"📥 Extrayendo datos desde: {api_url}")

try:
    response = requests.get(api_url)
    response.raise_for_status()  # Lanza un error si la petición falla (ej: 404)
    data = response.json()
    df_raw = pd.DataFrame(data)
    print(f"✅ ¡Extracción exitosa! Se cargaron {len(df_raw)} filas.")
    display(df_raw.head())

except requests.exceptions.RequestException as e:
    print(f"❌ Error al extraer los datos: {e}")
    df_raw = pd.DataFrame() # Creamos un dataframe vacío para evitar errores posteriores

except Exception as e:
    print(f"❌ Ocurrió un error inesperado: {e}")
    df_raw = pd.DataFrame()

📥 Extrayendo datos desde: https://www.datos.gov.co/resource/nudc-7mev.json?$limit=50000
✅ ¡Extracción exitosa! Se cargaron 14585 filas.


a_o c_digo_municipio   municipio c_digo_departamento departamento  \
0  2023            05004    Abriaquí                  05    Antioquia   
1  2023            95025  El Retorno                  95     Guaviare   
2  2023            95200  Miraflores                  95     Guaviare   
3  2023            97001        Mitú                  97       Vaupés   
4  2023            97161      Caruru                  97       Vaupés   

  c_digo_etc              etc poblaci_n_5_16 tasa_matriculaci_n_5_16  \
0       3758  Antioquia (ETC)            503                   62.62   
1       3830   Guaviare (ETC)           4438                   53.27   
2       3830   Guaviare (ETC)           2014                   32.52   
3       3831     Vaupés (ETC)          10986                   59.57   
4       3831     Vaupés (ETC)           1228                    51.3   

  cobertura_neta cobertura_neta_transici_n cobertura_neta_primaria  \
0          62.62                     44.19                   63.33   
1          53.27                     33.91                   48.89   
2          32.52                     17.58                   25.33   
3          59.57                     42.76                   55.95   
4           51.3                     76.32                   52.29   

  cobertura_neta_secundaria cobertura_neta_media cobertura_bruta  \
0                     51.53                40.23            66.8   
1                      44.9                 21.3           62.98   
2                     26.43                10.75           38.58   
3                     43.51                17.06           70.65   
4                     33.71                11.94           55.54   

  cobertura_bruta_transici_n cobertura_bruta_primaria  \
0                      58.14                    72.86   
1                       54.2                    65.19   
2                      36.36                    37.28   
3                       64.9                    76.96   
4                      92.11                    65.21   

  cobertura_bruta_secundaria cobertura_bruta_media deserci_n  \
0                      66.87                 56.32      1.19   
1                       69.6                 48.54      5.56   
2                       46.1                 26.16      7.85   
3                      72.92                 53.12      3.95   
4                      51.12                 27.36      8.36   

  deserci_n_transici_n deserci_n_primaria deserci_n_secundaria  \
0                    0               1.31                    0   
1                 6.95               4.99                 6.11   
2                   15               8.43                 6.36   
3                 2.27               1.84                 6.77   
4                 4.29               3.05                15.72   

  deserci_n_media aprobaci_n aprobaci_n_transici_n aprobaci_n_primaria  \
0            4.08      92.26                     0               96.73   
1            5.26      87.67                     0                87.9   
2            4.69      82.68                  3.33               84.64   
3            5.47      90.71                  0.57               94.12   
4           14.55       82.4                     0               89.63   

  aprobaci_n_secundaria aprobaci_n_media reprobaci_n reprobaci_n_transici_n  \
0                 83.49            93.88        6.55                      0   
1                  84.5            92.98        6.78                      0   
2                 79.51             87.5        9.47                   3.33   
3                 84.91            89.93        5.34                   0.57   
4                    69            78.18        9.24                      0   

  reprobaci_n_primaria reprobaci_n_secundaria reprobaci_n_media repitencia  \
0                 1.96                  16.51              2.04       9.52   
1                 7.11                   9.39              1.75       9.34   
2                 6.93 

In [3]:
df_raw

a_o c_digo_municipio    municipio c_digo_departamento departamento  \
0      2023            05004     Abriaquí                  05    Antioquia   
1      2023            95025   El Retorno                  95     Guaviare   
2      2023            95200   Miraflores                  95     Guaviare   
3      2023            97001         Mitú                  97       Vaupés   
4      2023            97161       Caruru                  97       Vaupés   
...     ...              ...          ...                 ...          ...   
14580  2011             5036  Angelópolis                   5    Antioquia   
14581  2011             5034        Andes                   5    Antioquia   
14582  2011             5031       Amalfi                   5    Antioquia   
14583  2011             5030        Amagá                   5    Antioquia   
14584  2011             5021   Alejandría                   5    Antioquia   

      c_digo_etc              etc poblaci_n_5_16 tasa_matriculaci_n_5_16  \
0           3758  Antioquia (ETC)            503                   62.62   
1           3830   Guaviare (ETC)           4438                   53.27   
2           3830   Guaviare (ETC)           2014                   32.52   
3           3831     Vaupés (ETC)          10986                   59.57   
4           3831     Vaupés (ETC)           1228                    51.3   
...          ...              ...            ...                     ...   
14580       3758  Antioquia (ETC)           1707                   78.85   
14581       3758  Antioquia (ETC)          10244                   84.45   
14582       3758  Antioquia (ETC)           5552                   97.71   
14583       3758  Antioquia (ETC)           6631                   78.65   
14584       3758  Antioquia (ETC)            702                  118.52   

      cobertura_neta cobertura_neta_transici_n cobertura_neta_primaria  \
0              62.62                     44.19                   63.33   
1              53.27                     33.91                   48.89   
2              32.52                     17.58                   25.33   
3              59.57                     42.76                   55.95   
4               51.3                     76.32                   52.29   
...              ...                       ...                     ...   
14580           78.9                      56.2                    82.4   
14581           84.5                      65.1                    83.7   
14582           97.7                      61.7                   107.2   
14583           78.7                      55.3                    76.3   
14584          118.5                      82.7                   127.6   

      cobertura_neta_secundaria cobertura_neta_media cobertura_bruta  \
0                         51.53                40.23            66.8   
1                          44.9                 21.3           62.98   
2                         26.43                10.75           38.58   
3                         43.51                17.06           70.65   
4                         33.71                11.94           55.54   
...                         ...                  ...             ...   
14580                      59.9                 26.4            82.8   
14581                        62                 32.2            91.7   
14582                      65.1                 27.3           108.4   
14583                      70.7                 38.9            87.7   
14584                      90.2                 48.5           134.9   

      cobertura_bruta_transici_n cobertura_bruta_primaria  \
0                          58.14                    72.86   
1                           54.2                    65.19   
2                          36.36                    37.28   
3                           64.9                    76.96   
4                          92.11                    65.21   
...                          ...      

TRANSFORMACIÓN - LIMPIEZA Y NORMALIZACIÓN

In [4]:
conn = sqlite3.connect('../Datos/Educacion.bd')
df_raw.to_sql('Educacion', conn, if_exists='replace', index=False)

14585

In [5]:
def ejecutar_sql(query, conexion=conn):
    """
    Ejecuta una consulta SQL. Si es un SELECT, muestra el resultado como tabla.
    Si no, ejecuta sin retorno (DROP, CREATE, etc.).
    """
    if query.strip().lower().startswith("select"):
        df = pd.read_sql_query(query, conexion)
        display(df)
    else:
        cursor = conexion.cursor()
        cursor.execute(query)
        conexion.commit()
        print("✅ Consulta ejecutada correctamente.")

Crear tabla limpia

In [58]:
ejecutar_sql("DROP TABLE IF EXISTS Educacion_Limpia")

✅ Consulta ejecutada correctamente.


In [59]:
ejecutar_sql("""
CREATE TABLE Educacion_Limpia AS
SELECT
    CAST(a_o AS INTEGER) AS anio,
    
    -- Normalizamos el código departamento (ej: '1' → '01')
    CASE 
        WHEN LENGTH(TRIM(c_digo_departamento)) = 1 THEN '0' || TRIM(c_digo_departamento)
        ELSE TRIM(c_digo_departamento)
    END AS codigo_departamento,
    
    -- Normalizamos el código municipio (ej: '5400' → '05400')
    CASE 
        WHEN LENGTH(TRIM(c_digo_municipio)) = 4 THEN '0' || TRIM(c_digo_municipio)
        ELSE TRIM(c_digo_municipio)
    END AS codigo_municipio,
    
    -- Para trazabilidad dejamos nombres también
    TRIM(departamento) AS nombre_departamento,
    TRIM(municipio) AS nombre_municipio,
    
    -- Conversión de métricas
    CAST(poblaci_n_5_16 AS REAL) AS poblacion,
    CAST(tasa_matriculaci_n_5_16 AS REAL) AS tasa_matriculacion,
    CAST(cobertura_neta AS REAL) AS cobertura_neta,
    CAST(aprobaci_n AS REAL) AS aprobacion,
    CAST(reprobaci_n AS REAL) AS reprobacion
    
FROM Educacion
WHERE 
    a_o IS NOT NULL
    AND TRIM(departamento) != 'NACIONAL'
    AND c_digo_departamento IS NOT NULL
    AND c_digo_municipio IS NOT NULL
    AND poblaci_n_5_16 IS NOT NULL
    AND tasa_matriculaci_n_5_16 IS NOT NULL
    AND cobertura_neta IS NOT NULL;
""")


✅ Consulta ejecutada correctamente.


In [60]:
pd.read_sql('SELECT * FROM Educacion_Limpia', con=conn)

anio codigo_departamento codigo_municipio nombre_departamento  \
0      2023                  05            05004           Antioquia   
1      2023                  95            95025            Guaviare   
2      2023                  95            95200            Guaviare   
3      2023                  97            97001              Vaupés   
4      2023                  97            97161              Vaupés   
...     ...                 ...              ...                 ...   
14443  2011                  05            05036           Antioquia   
14444  2011                  05            05034           Antioquia   
14445  2011                  05            05031           Antioquia   
14446  2011                  05            05030           Antioquia   
14447  2011                  05            05021           Antioquia   

      nombre_municipio  poblacion  tasa_matriculacion  cobertura_neta  \
0             Abriaquí      503.0               62.62           62.62   
1           El Retorno     4438.0               53.27           53.27   
2           Miraflores     2014.0               32.52           32.52   
3                 Mitú    10986.0               59.57           59.57   
4               Caruru     1228.0               51.30           51.30   
...                ...        ...                 ...             ...   
14443      Angelópolis     1707.0               78.85           78.90   
14444            Andes    10244.0               84.45           84.50   
14445           Amalfi     5552.0               97.71           97.70   
14446            Amagá     6631.0               78.65           78.70   
14447       Alejandría      702.0              118.52          118.50   

       aprobacion  reprobacion  
0           92.26         6.55  
1           87.67         6.78  
2           82.68         9.47  
3           90.71         5.34  
4           82.40         9.24  
...           ...          ...  
14443       86.27         5.59  
14444       95.46         0.58  
14445       91.25         0.00  
14446       86.86         8.99  
14447       92.39         3.58  

[14448 rows x 10 columns]

CREACIÓN DE DIMENSIONES

In [61]:
ejecutar_sql("DROP TABLE IF EXISTS Dim_Tiempo;")

✅ Consulta ejecutada correctamente.


In [62]:
ejecutar_sql("""
CREATE TABLE Dim_Tiempo AS
SELECT anio,
    ROW_NUMBER() OVER (ORDER BY anio) AS id_tiempo
FROM (
    SELECT DISTINCT anio
    FROM Educacion_Limpia
    WHERE anio IS NOT NULL
)
""")


✅ Consulta ejecutada correctamente.


In [63]:
pd.read_sql('SELECT * FROM Dim_Tiempo;', con=conn)

anio  id_tiempo
0   2011          1
1   2012          2
2   2013          3
3   2014          4
4   2015          5
5   2016          6
6   2017          7
7   2018          8
8   2019          9
9   2020         10
10  2021         11
11  2022         12
12  2023         13

In [64]:
ejecutar_sql("DROP TABLE IF EXISTS Dim_Geografia;")

✅ Consulta ejecutada correctamente.


In [65]:
ejecutar_sql("""
CREATE TABLE Dim_Geografia AS
SELECT 
    codigo_departamento,
    codigo_municipio,
    MIN(nombre_departamento) AS nombre_departamento,
    MIN(nombre_municipio) AS nombre_municipio,
    ROW_NUMBER() OVER (ORDER BY codigo_departamento, codigo_municipio) AS id_geografia
FROM Educacion_Limpia
GROUP BY codigo_departamento, codigo_municipio;
""")

✅ Consulta ejecutada correctamente.


In [66]:
pd.read_sql('SELECT * FROM Dim_Geografia;', con=conn)

codigo_departamento codigo_municipio nombre_departamento  \
0                     05            05001           Antioquia   
1                     05            05002           Antioquia   
2                     05            05004           Antioquia   
3                     05            05021           Antioquia   
4                     05            05030           Antioquia   
...                  ...              ...                 ...   
1117                  97            97889              Vaupés   
1118                  99            99001             Vichada   
1119                  99            99524             Vichada   
1120                  99            99624             Vichada   
1121                  99            99773             Vichada   

     nombre_municipio  id_geografia  
0            Medellín             1  
1           Abejorral             2  
2            Abriaquí             3  
3          Alejandría             4  
4               Amagá             5  
...               ...           ...  
1117         Yavaraté          1118  
1118   Puerto Carreño          1119  
1119     La Primavera          1120  
1120    Santa Rosalía          1121  
1121         Cumaribo          1122  

[1122 rows x 5 columns]

CREACIÓN DE LA TABLA DE HECHOS

In [67]:
ejecutar_sql("DROP TABLE IF EXISTS Fact_Matriculas;")

✅ Consulta ejecutada correctamente.


In [68]:
ejecutar_sql("""
CREATE TABLE Fact_Matriculas AS
SELECT 
    tg.id_tiempo,
    gg.id_geografia,
    el.poblacion,
    el.tasa_matriculacion,
    el.cobertura_neta,
    el.aprobacion,
    el.reprobacion,
    ROUND(el.poblacion * el.tasa_matriculacion / 100.0) AS total_matriculados
FROM Educacion_Limpia el
JOIN Dim_Tiempo tg ON el.anio = tg.anio
JOIN Dim_Geografia gg ON el.codigo_departamento = gg.codigo_departamento AND el.codigo_municipio = gg.codigo_municipio
""")

✅ Consulta ejecutada correctamente.


In [69]:
pd.read_sql('SELECT * FROM Fact_Matriculas;', con=conn)

id_tiempo  id_geografia  poblacion  tasa_matriculacion  cobertura_neta  \
0             13             3      503.0               62.62           62.62   
1             13          1111     4438.0               53.27           53.27   
2             13          1112     2014.0               32.52           32.52   
3             13          1113    10986.0               59.57           59.57   
4             13          1114     1228.0               51.30           51.30   
...          ...           ...        ...                 ...             ...   
14443          1             8     1707.0               78.85           78.90   
14444          1             7    10244.0               84.45           84.50   
14445          1             6     5552.0               97.71           97.70   
14446          1             5     6631.0               78.65           78.70   
14447          1             4      702.0              118.52          118.50   

       aprobacion  reprobacion  total_matriculados  
0           92.26         6.55               315.0  
1           87.67         6.78              2364.0  
2           82.68         9.47               655.0  
3           90.71         5.34              6544.0  
4           82.40         9.24               630.0  
...           ...          ...                 ...  
14443       86.27         5.59              1346.0  
14444       95.46         0.58              8651.0  
14445       91.25         0.00              5425.0  
14446       86.86         8.99              5215.0  
14447       92.39         3.58               832.0  

[14448 rows x 8 columns]

CONSULTAS Y RESPUESTAS

¿Qué porcentaje de escolaridad hay?

In [70]:
print("Porcentaje de escolaridad a nivel nacional:")
ejecutar_sql("""
SELECT 
    ROUND(SUM(total_matriculados)*1.0 / SUM(poblacion)*100, 2) AS escolaridad_promedio_pct
FROM Fact_Matriculas;
""")

Porcentaje de escolaridad a nivel nacional:


escolaridad_promedio_pct
0                     88.59

Porcentaje de escolaridad por año

In [71]:
print("Porcentaje de escolaridad por año:")
ejecutar_sql("""
SELECT 
    t.anio,
    ROUND(SUM(f.total_matriculados) * 1.0 / SUM(f.poblacion) * 100, 2) AS escolaridad_promedio_pct
FROM Fact_Matriculas f
JOIN Dim_Tiempo t ON f.id_tiempo = t.id_tiempo
GROUP BY t.anio
ORDER BY t.anio;
""")


Porcentaje de escolaridad por año:


anio  escolaridad_promedio_pct
0   2011                     90.45
1   2012                     88.28
2   2013                     87.50
3   2014                     86.11
4   2015                     85.86
5   2016                     85.58
6   2017                     85.17
7   2018                     88.07
8   2019                     92.55
9   2020                     92.17
10  2021                     83.20
11  2022                     91.72
12  2023                     90.46

Porcentaje de escolaridad por departamento

In [72]:
print("📍 Porcentaje de escolaridad por departamento:")
ejecutar_sql("""
SELECT 
    g.nombre_departamento,
    ROUND(SUM(f.total_matriculados) * 1.0 / SUM(f.poblacion) * 100, 2) AS escolaridad_promedio_pct
FROM Fact_Matriculas f
JOIN Dim_Geografia g ON f.id_geografia = g.id_geografia
GROUP BY g.nombre_departamento
ORDER BY escolaridad_promedio_pct DESC;
""")


📍 Porcentaje de escolaridad por departamento:


nombre_departamento  \
0                                            Casanare   
1                                               Sucre   
2                                           Santander   
3                                           Magdalena   
4                                               Cesar   
5                                             Bolívar   
6                                                Meta   
7                                        Cundinamarca   
8                                           Risaralda   
9                                             Córdoba   
10                                          Antioquia   
11                                          Atlántico   
12                                             Tolima   
13                                 Norte de Santander   
14                                              Huila   
15                                             Boyacá   
16                                            Quindio   
17                                        Bogotá D.C.   
18                                              Cauca   
19                                           Amazonas   
20                                    Valle del Cauca   
21                                         La Guajira   
22                                              Chocó   
23                                             Caldas   
24                                            Caquetá   
25                                            Guainía   
26                                             Arauca   
27                                           Putumayo   
28                                             Nariño   
29  Archipiélago de San Andrés, Providencia y Sant...   
30                                            Vichada   
31                                           Guaviare   
32                                             Vaupés   

    escolaridad_promedio_pct  
0                      98.92  
1                      96.60  
2                      94.93  
3                      94.39  
4                      93.99  
5                      93.22  
6                      92.79  
7                      92.14  
8                      92.11  
9                      91.16  
10                     91.15  
11                     90.96  
12                     89.96  
13                     89.54  
14                     89.48  
15                     87.89  
16                     86.79  
17                     86.61  
18                     84.45  
19                     83.76  
20                     83.29  
21                     82.93  
22                     82.88  
23                     81.47  
24                     81.11  
25                     80.07  
26                     79.37  
27                     79.13  
28                     77.37  
29                     75.09  
30                     70.19  
31                     63.99  
32                     60.52

Porcentaje de escolaridad por año y departamento

In [74]:
print("Porcentaje de escolaridad por año y departamento:")
ejecutar_sql("""
SELECT 
    t.anio,
    g.nombre_departamento,
    ROUND(SUM(f.total_matriculados) * 1.0 / SUM(f.poblacion) * 100, 2) AS escolaridad_promedio_pct
FROM Fact_Matriculas f
JOIN Dim_Tiempo t ON f.id_tiempo = t.id_tiempo
JOIN Dim_Geografia g ON f.id_geografia = g.id_geografia
GROUP BY t.anio, g.nombre_departamento
ORDER BY t.anio, escolaridad_promedio_pct DESC;
""")


Porcentaje de escolaridad por año y departamento:


anio nombre_departamento  escolaridad_promedio_pct
0    2011               Sucre                    103.27
1    2011            Casanare                    100.68
2    2011           Atlántico                     98.12
3    2011           Risaralda                     95.61
4    2011               Cesar                     94.75
..    ...                 ...                       ...
424  2023            Guaviare                     77.56
425  2023            Amazonas                     75.76
426  2023             Guainía                     71.61
427  2023             Vichada                     67.63
428  2023              Vaupés                     55.17

[429 rows x 3 columns]

### Escolaridad Promedio Nacional

El porcentaje de escolaridad promedio en Colombia para el periodo completo fue de 88.59%, lo que indica que cerca de 9 de cada 10 niños entre 5 y 16 años están matriculados en el sistema educativo.

Este valor refleja un nivel relativamente alto de cobertura educativa a nivel nacional, aunque aún hay margen de mejora, especialmente en algunas regiones específicas.

### Escolaridad por Año (2011–2023)

Tendencia general: Se observa una ligera disminución entre 2011 (90.45%) y 2017 (85.17%), seguida de una recuperación sostenida entre 2018 y 2020, donde se superan incluso los niveles iniciales.

Años más altos: Los mejores años fueron 2019 (92.55%), 2020 (92.17%) y 2022 (91.72%), lo que sugiere una mejora en las estrategias de cobertura en esos periodos.

Año más bajo: 2021 tuvo el menor porcentaje (83.20%), lo cual podría estar asociado a los efectos prolongados de la pandemia de COVID-19, que afectó gravemente la continuidad escolar.

### Escolaridad por Departamento

Departamentos destacados:

Casanare lidera el ranking con una cobertura de 98.92%, seguido de Sucre (96.60%) y Santander (94.93%).

Esto sugiere una gestión eficaz de la cobertura educativa en estos territorios, probablemente asociada a esfuerzos focalizados o menor dispersión geográfica.

Departamentos con menor cobertura:

Vaupés (60.52%), Guaviare (63.99%), y Vichada (70.19%) tienen los niveles más bajos. Estas regiones tienen condiciones geográficas y de acceso más difíciles, lo cual puede afectar la escolarización.

También se observa baja cobertura en territorios amazónicos e insulares como Amazonas, Guainía, y San Andrés, lo que indica brechas regionales significativas.



¿Cómo compararía el rendimiento educativo por municipios? (aprobación vs. reprobación)

In [83]:
print("Rendimiento educativo promedio por municipio (Top 10 con mayor aprobación):")
ejecutar_sql("""
SELECT 
    g.nombre_departamento, 
    g.nombre_municipio,
    ROUND(AVG(aprobacion), 2) AS aprobacion_promedio,
    ROUND(AVG(reprobacion), 2) AS reprobacion_promedio
FROM Fact_Matriculas f
JOIN Dim_Geografia g ON f.id_geografia = g.id_geografia
GROUP BY g.nombre_departamento, g.nombre_municipio
HAVING aprobacion_promedio IS NOT NULL
ORDER BY aprobacion_promedio DESC
limit 10;
""")


Rendimiento educativo promedio por municipio (Top 10 con mayor aprobación):


nombre_departamento nombre_municipio  aprobacion_promedio  \
0            Amazonas      La Victoria               100.00   
1            Amazonas     Puerto Arica                98.97   
2        Cundinamarca         Tibirita                98.42   
3        Cundinamarca         La Palma                98.28   
4        Cundinamarca        Guatavita                98.27   
5        Cundinamarca         Chaguaní                98.19   
6              Nariño         Pupiales                98.10   
7            Amazonas  Miriti - Paraná                98.06   
8        Cundinamarca        Sutatausa                97.96   
9           Santander          Cabrera                97.90   

   reprobacion_promedio  
0                  0.00  
1                  0.03  
2                  0.43  
3                  0.20  
4                  0.27  
5                  0.16  
6                  0.67  
7                  0.38  
8                  0.11  
9                  0.95

¿Qué departamentos tienen mejor cobertura?

In [85]:
print("Departamentos con mejor cobertura neta promedio (Top 10):")
ejecutar_sql("""
SELECT 
    g.nombre_departamento,
    ROUND(AVG(cobertura_neta), 2) AS cobertura_promedio
FROM Fact_Matriculas f
JOIN Dim_Geografia g ON f.id_geografia = g.id_geografia
GROUP BY g.nombre_departamento
HAVING cobertura_promedio IS NOT NULL
ORDER BY cobertura_promedio DESC
LIMIT 10;
""")


Departamentos con mejor cobertura neta promedio (Top 10):


nombre_departamento  cobertura_promedio
0             Quindio               94.58
1               Sucre               93.85
2               Cesar               93.81
3           Magdalena               93.28
4                Meta               90.74
5         Bogotá D.C.               89.49
6              Tolima               89.24
7            Casanare               88.58
8        Cundinamarca               88.55
9           Antioquia               88.11

Todos los departamentos del top 10 tienen una cobertura neta promedio superior al 88%, lo que indica un acceso escolar relativamente alto entre la población en edad escolar (5 a 16 años).

Quindío (94.58%), Sucre (93.85%) y Cesar (93.81%) lideran la lista con coberturas sobresalientes.
El Quindío se posiciona como el departamento con mayor cobertura neta promedio (94.58%), superando incluso a grandes centros urbanos.
Bogotá D.C. ocupa la posición 5, con una cobertura del 89.49%. A pesar de ser la capital, no lidera, lo que podría estar relacionado con:

Mayor deserción escolar en zonas vulnerables.

Migración interna que desborda la capacidad de atención.

Retos de integración de población migrante (ej. población venezolana).

In [86]:
# Cerramos la conexión
conn.close()
print("Conexión a la base de datos cerrada.")

Conexión a la base de datos cerrada.
